### ISSUE TITLE
Right aligned blocks: toolbar can be very far away from inner content

### ISSUE THREAD

**Issue by creativecoder** [WordPress/gutenberg#37515]
### Description

When a block's content is right aligned, the block toolbar can be very far away from the inner content on wider screens.

I saw this trip up new users in recent testing with the navigation block, which is often right aligned in the header.

### Step-by-step reproduction instructions

- With a block based theme that has a navigation block in the header, set the nav item justification to be right aligned.
- Click on the navigation block content and notice the toolbar position.

### Screenshots, screen recording, code snippet

<img width="1203" alt="image" src="https://user-images.githubusercontent.com/1699996/146697581-a0e67571-cbc0-4023-b67d-387ea9c09151.png">

Toolbar is far left, with the inner blocks on the far right.

### Environment info

- WP v5.8.2
- Gutenberg v12.2.0-rc.1

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

No

Related: https://github.com/WordPress/gutenberg/issues/36701

**Comment by jasmussen** [comment]
Thanks for the issue. Would you say this is similar to the Row or Paragraph blocks?

<img width="919" alt="row block justified right" src="https://user-images.githubusercontent.com/1204802/146905516-a0b98fb3-7e6b-4695-be8e-cb199a76eee2.png">

<img width="907" alt="paragraph aligned right" src="https://user-images.githubusercontent.com/1204802/146905552-27985cd3-cc8e-48c9-9222-d3666bca4994.png">

**Comment by talldan** [comment]
I agree with @jasmussen. This is more of a general block UI issue rather than a navigation block issue.

**Comment by creativecoder** [comment]
>  Would you say this is similar to the Row or Paragraph blocks?

Yes! I hadn't noticed it with those other blocks before, but you're right, it's not specific to the Navigation block. I'll update the issue title and description.
